### ISSUE TITLE
zbus 4.2.0 breaks server-side handshake on FreeBSD

### ISSUE THREAD

**Issue by jbeich** [dbus2/zbus#756]
Regressed by edc79a338b74. Found via dbus2/busd@153fe2b7e0a9.

```
$ busd &
$ export DBUS_SESSION_BUS_ADDRESS=unix:path=$XDG_RUNTIME_DIR/busd-session
$ basuctl --user
<27>(../src/busctl/busctl.c:154) Failed to list names: Connection reset by peer
$ dbus-send --print-reply --session --type=method_call --dest=org.freedesktop.DBus / org.freedesktop.DBus.Peer.Ping
Failed to open connection to "session" message bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
```

**Comment by zeenix** [comment]
@jbeich Could you check for me if #757 really fixes the zbus tests on FreeBSD? You can disabled the systemd test by appending `-- --skip fdpass_systemd` to the `cargo test` command. Once you've verified, I can merge and roll out the release.

**Comment by jbeich** [comment]
I confirm, after applying the fix tests pass like in zbus-4.1.2:
```rust
$ cargo test --features bus-impl -- --skip fdpass_systemd
[...]
---- connection::p2p_tests::tcp_p2p stdout ----
thread '<unnamed>' panicked at zbus/src/connection/mod.rs:1450:48:
called `Result::unwrap()` on an `Err` value: Handshake("Could not send zero byte with credentials")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'connection::p2p_tests::tcp_p2p' panicked at zbus/src/connection/mod.rs:1448:5:
explicit panic


failures:
    connection::p2p_tests::tcp_p2p

test result: FAILED. 38 passed; 1 failed; 3 ignored; 0 measured; 1 filtered out; finished in 0.12s

error: test failed, to rerun pass `-p zbus --lib`
```

**Comment by zeenix** [comment]
> I confirm, after applying the fix tests pass like in zbus-4.1.2

Great, thanks!

> ```rust
> thread 'connection::p2p_tests::tcp_p2p' panicked at zbus/src/connection/mod.rs:1448:5:
> ```

Oh, I didn't realize that was broken on FreeBSD. I'll try to fix that and hopefully I can ask you again to test. :)
